texar/modules/classifiers/conv_classifiers.py

```diff
@@ -241,3 +241,14 @@ def layer_names(self) -> List[str]:
         r"""A list of uniquified layer names.
         """
         return self._encoder.layer_names
+
+    @property
+    def output_size(self) -> int:
+        r"""Return the final dimension of the output :attr:`logit`,
+        if the final dimension cannot be computed, returns 1.
```

Comment:
Do we return `-1` here? Now if `self._hparams.num_classes < 1` we would just run into a `NameError` because `logit_dim` would not be defined.